<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>815</width>
    <height>753</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="stopButton">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>190</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Stop</string>
    </property>
   </widget>
   <widget class="QPushButton" name="forwardButton">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>0</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Forward</string>
    </property>
   </widget>
   <widget class="QPushButton" name="backwardButton">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>40</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Backward</string>
    </property>
   </widget>
   <widget class="QPushButton" name="leftButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>t-Left</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rightButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>20</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>t-Right</string>
    </property>
   </widget>
   <widget class="QPushButton" name="followLineButton">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>20</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>FOLLOW</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>0</y>
      <width>321</width>
      <height>311</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="4" column="1">
      <widget class="QLineEdit" name="lineEdit_speedB"/>
     </item>
     <item row="5" column="1">
      <widget class="QLineEdit" name="lineEdit_verticalDistance"/>
     </item>
     <item row="3" column="0">
      <widget class="QCheckBox" name="checkBox_readEncoderB">
       <property name="text">
        <string>encoderRight</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QCheckBox" name="checkBox_readColorSensor">
       <property name="text">
        <string>color</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="lineEdit_encoderB"/>
     </item>
     <item row="5" column="0">
      <widget class="QCheckBox" name="checkBox_readSonicSensor">
       <property name="text">
        <string>Distance</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QLineEdit" name="lineEdit_colorSensor"/>
     </item>
     <item row="7" column="1">
      <widget class="QLineEdit" name="lineEdit_lineSensorA"/>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>sensor</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>value</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>    speedRight(red)</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QCheckBox" name="checkBox_readLineSensorA">
       <property name="text">
        <string>linesensorLeft</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QCheckBox" name="checkBox_readEncoderA">
       <property name="text">
        <string>encoderLeft</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="lineEdit_speedA"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>    speedLeft(blue)</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="lineEdit_encoderA"/>
     </item>
     <item row="8" column="0">
      <widget class="QCheckBox" name="checkBox_readLineSensorB">
       <property name="text">
        <string>linesensorRight</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QLineEdit" name="lineEdit_lineSensorB"/>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="Odometrie">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>190</y>
      <width>91</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Odometrie</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>72</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QCustomPlot" name="customPlot" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>310</y>
      <width>761</width>
      <height>351</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>231</width>
      <height>131</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="3" column="2">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_KdLeft"/>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Kp</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_KpLeft"/>
     </item>
     <item row="1" column="3">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_KpRight"/>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Ki</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Kd</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_KiLeft"/>
     </item>
     <item row="2" column="3">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_KiRight"/>
     </item>
     <item row="0" column="3">
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Right</string>
       </property>
      </widget>
     </item>
     <item row="3" column="3">
      <widget class="QDoubleSpinBox" name="doubleSpinBox_KdRight"/>
     </item>
     <item row="0" column="2">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Left</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>79</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Speed Set</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>230</y>
      <width>67</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Path</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_path">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>260</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_path">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>230</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>load</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ChallengeStartButton">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>230</y>
      <width>91</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Challenge</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_verticalDistance_2">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>640</y>
      <width>178</width>
      <height>29</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_workphase">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>280</y>
      <width>171</width>
      <height>29</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Work phase</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>815</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
